### DISCUSSION TITLE
property missing in bun but works in node

### DISCUSSION THREAD

**Discussion by wenerme** [mikro-orm/mikro-orm#6005] [General]
### Describe the bug

xref #5986 
xref https://github.com/oven-sh/bun/issues/13766

> Trying to query by not existing property Test1Entity.notes

I tried copy the mixin function in to project(e.g. copy withCodeEntity to project), the property will works fine.
I checked the Entity is reference equal to current project, 

```ts
import { checkMikroOrmEnv } from '@wener/nestjs/entity';

checkMikroOrmEnv({
  Entity,
  MetadataStorage,
});
```

### Reproduction

```bash
git clone https://github.com/wenerme/mikro-orm-issues
cd mikro-orm-issues
pnpm i

# failed with missing prop
make bun-run-is-1

# failed with missing table, means property exists, just works
make tsx-run-is-1
```

### What driver are you using?

None

### MikroORM version

6.3.8

### Node.js version

20.16.0

### Operating system

macos

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by wenerme** [comment]
Hi @B4nan, I noticed this was moved to Discussions, and I was wondering if you could clarify the reason for the move? If MikroORM doesn’t officially support Bun, that’s completely fine with me, I just wanted to confirm. Thanks!

**Reply by B4nan** [reply]
> property missing in bun but works in node

This is by definition something to report to bun folks (which you already did, so let's see what they say), not here, I don't plan to deal with their incompatibilities. 

This is not about the ORM supporting bun, it's about them not supporting node the way they claim to be. What I would call unsupported are the mixins, which is the second reason why I am not really eager to debug this at all. 

Also, the fact that you need all this magic is suspicious at best, how do you know the issue is not in your code if you need this? Aren't you changing the class names there actually?

https://github.com/wenerme/mikro-orm-issues/blob/main/src/patchMikroORMMetadataStorage.ts

**Reply by wenerme** [reply]
Hi @B4nan, we previously discussed the mixin issue here: [link to discussion](https://github.com/mikro-orm/mikro-orm/discussions/5371). 

While I’m hoping for a better way to handle the entity name conflict, my current workaround involves a script like `patchMikroORMMetadataStorage.ts`. Essentially, I’ve had to modify the class name to trick MikroORM, even though the class names are identical but the properties might differ. If there's a more efficient approach or future plans to address this, I'd love to know. Thanks!

**Reply by B4nan** [reply]
If you ask me, I would just forget about mixins, or use `EntitySchema` instead of decorators, where you can compose the properties in any way you want.

**Reply by wenerme** [reply]
BTW, I’d love to share some of my work with you! I've created several small mixins, which you can check out here: [wenerme/wode/tree/main/packages/nestjs/src/entity/mixins](https://github.com/wenerme/wode/tree/main/packages/nestjs/src/entity/mixins). I also use mixins with `type-graphql`. By leveraging mixins, I can extract common business logic and ensure consistent patterns across many entities, which has significantly streamlined my workflow.

**Reply by wenerme** [reply]
@B4nan I see your point. To me, EntitySchema feels a bit more like how Sequelize handles things—maybe I haven’t explored it deeply enough. Personally, I prefer working with classes, as it feels more like a traditional ORM experience. I find mixins to be a useful pattern for encapsulating common logic across entities, but I’ll definitely give EntitySchema another look. Thanks for the suggestion!
